javascript/extractor/src/com/semmle/jcorn/flow/FlowParser.java

```diff
@@ -78,6 +77,8 @@ protected Token getTokenFromCode(int code) {
 		private Function<Token, Void> onToken;
 		private boolean inType, noAnonFunctionType;
 		private List<Token> tokens = new ArrayList<Token>();
+		private Function<SyntaxError, ?> onRecoverableError;
```

Comment:
`Function<SyntaxError, Void>`, like for `onToken` above?

Reply:
Alternatively, both could be changed to have type [`Consumer`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html).

Reply:
Changing `?` to `Void` does not, unfortunately, work, since there is one place where we pass in a `Function<SyntaxError, Boolean>`.